crates/ruma-identity-service-api/src/association/unbind_3pid/v2.rs

```diff
@@ -0,0 +1,67 @@
+//! [POST /_matrix/identity/v2/3pid/unbind](https://matrix.org/docs/spec/identity_service/r0.3.0#post-matrix-identity-v2-3pid-unbind)
+
+use ruma_api::ruma_api;
+use ruma_common::thirdparty::Medium;
+use ruma_identifiers::user_id::UserId;
+use serde::{Deserialize, Serialize};
+
+ruma_api! {
+    metadata: {
+        description: "Remove an association between a session and a Matrix user ID.",
+        method: POST,
+        name: "unbind_3pid",
+        path: "/_matrix/identity/v2/3pid/unbind",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    request: {
+        /// The Session ID generated by the `requestToken` call.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub sid: Option<&'a str>,
+
+        /// The client secret passed to the `requestToken` call.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub client_secret: Option<&'a str>,
```

Comment:
Looking at the response in the clarification issue, this is a second layer of authentication, and the signing scheme is the same as the federation API.

Since there are only 2 options to authenticate the request as coming from the owner of the 3PID: either by signing the request, or by providing the `sid`/`client_secret` pair. I think it would make sense to bundle the pair into one structure. That's why I created the `ThreePidOwnershipProof` struct (which might need a name change, I just couldn't find a proper name for it). It's wrapped inside an `Option`, and the comment clearly state that if `None` is used for the field, the request must then be signed by the appropriate homeserver. 
We could also create a separate enum, but it seems that it would be analogous to using `Option` (unless I misunderstood how requests are signed in `ruma-federation-api`).